runtime/jvmti/jvmtiStackFrame.cpp

```diff
@@ -905,13 +905,89 @@ jvmtiInternalGetStackTrace(
 }
 
 #if JAVA_SPEC_VERSION >= 25
+/**
+ * @brief Iterator function used during a stack walk to clear frame pop requests.
+ *
+ * This function is called for each frame during a stack walk. It clears the
+ * J9SF_A0_REPORT_FRAME_POP_TAG flag from the frame's base pointer to prevent
+ * JVMTI FramePop events from being generated.
+ *
+ * @param[in] currentThread The current VM thread performing the stack walk.
+ * @param[in,out] walkState The stack walk state, containing the frame pointer.
+ *
+ * @return J9_STACKWALK_KEEP_ITERATING Always returns keep iterating to continue walking all frames.
+ */
+static UDATA
+clearAllFramePopsIterator(J9VMThread *currentThread, J9StackWalkState *walkState)
+{
+	*walkState->bp &= ~(UDATA)J9SF_A0_REPORT_FRAME_POP_TAG;
```

Comment:
Does there need to be a check for interpreted frame here? I believe the BP can point to the return address slot in jitted frames, so the mask will destroy the RA on some platforms.